lib/java/ydoc-server/src/main/java/org/enso/ydoc/polyfill/web/Timers.java

```diff
@@ -92,7 +84,17 @@ public Object execute(Value... arguments) {
     };
   }
 
-  private Runnable execute(Value func, Object[] args) {
+  private Runnable runnable(Value func, Object[] args) {
     return () -> executor.execute(() -> func.executeVoid(args));
   }
+
+  /**
+   * Applies Node.js constraints on a delay value.
+   *
+   * @param delay the input delay value.
+   * @return the delay value bounded to the permitted range.
+   */
+  private long normalizedDelay(long delay) {
+    return Math.min(MAXIMUM_DELAY, Math.max(MINIMUM_DELAY, delay));
```

Comment:
What are "Node.js constraints"? According to [MDN setInterval](https://developer.mozilla.org/en-US/docs/Web/API/setInterval) defaults to zero if not specified. 

I'd say `MAXIMUM_DELAY` being `1` is _too much_ and it changes semantics. We want to support _delay 0_ to run as soon as possible once current (event queue) task is over.

Reply:
MDN specifies a minimum delay for browsers (4ms) https://developer.mozilla.org/en-US/docs/Web/API/setInterval#delay_restrictions

Node.js specifies both minimum and maximum delay values https://nodejs.org/api/timers.html#setintervalcallback-delay-args